### DISCUSSION TITLE
Provide more insights on adoption of OSS projects 📊

### DISCUSSION THREAD

**Discussion by tomkerkhove** [community/community#4660] [Other Feature Feedback, Questions, & Ideas]
Provide more insights on adoption of OSS projects as this is very hard to track.

What I'm mainly looking for:

- Who is using my project? Which companies are they?
- How are our artifacts doing and what's the adoption of that? (See https://github.com/github/feedback/discussions/4658)
- ...

For me the "Who is using my project" and bringing Docker Hub metrics on my GitHub repo to show it over time would be killer features.

## Backstory
There are ton of info on who is contributing and so on; but what about the adoption?

I started looking at this because Microsoft was wondering what the adoption of Promitor is which I run and actually, I was pretty clueless. I wrote a project summary which included some information which I was looking for but measuring adoption in general is a big gap which would be nice to get help from GitHub

So I've included a bit of star growth (https://star-history.t9t.io/#tomkerkhove/promitor), contributor info (already built-in, thanks) and Docker image pulls.

What I saw that would be nice to have is:
- https://github.com/github/feedback/discussions/4659
- Surface artifact metrics in the repo itself for potential new users. I use Docker images so GitHub could scrape Docker Hub metrics and report them on my repo (same can be done for NuGet, et al). Badges help here but they don't show growth over time and I had to write a scraper myself to Application Insights (which contributers can't see - https://github.com/tomkerkhove/docker-hub-metrics-scraper)
- Insights of # contributions per company (based on contributing users profile)
- Star growth (optionally with more info such as companies, geo, etc) 

## Examples
CNCF did a ton of work on this and actually spins up metrics for every project which can be used as inspiration - https://keda.devstats.cncf.io/d/8/dashboards?orgId=1&refresh=15m

Relates to https://github.com/github/feedback/discussions/4655 which could be one of the discovery options on how to fetch metrics on the artifact level.

**Comment by tomkerkhove** [comment]
Here's a mock-up:
![Picture1](https://user-images.githubusercontent.com/4345663/136081943-d43d512c-eb03-4a23-b5b6-3bfa9cacb3e1.png)

![Picture3](https://user-images.githubusercontent.com/4345663/136082083-cf20d680-d8a7-4d58-ae89-02b0f0ada81f.png)

**Reply by ghost** [reply]
Hi! So... I added a part of sponsors, hope to help.
![image](https://user-images.githubusercontent.com/95204402/167536589-2572c0aa-7fe1-47d6-b231-f16e1224730e.png)
